wild_lib/src/elf_writer.rs

```diff
@@ -1661,6 +1735,79 @@ fn apply_relocation(
     Ok(next_modifier)
 }
 
+fn apply_debug_relocation(
+    object_layout: &ObjectLayout,
+    offset_in_section: u64,
+    rel: &elf::Rela,
+    section_info: SectionInfo,
+    layout: &Layout,
+    out: &mut [u8],
+) -> Result<()> {
+    let section_address = section_info.section_address;
+    let place = section_address + offset_in_section;
+    let _span = tracing::trace_span!("debug relocation", address = place).entered();
+
+    let e = LittleEndian;
+    let symbol_index = rel
+        .symbol(e, false)
+        .context("Unsupported absolute relocation")?;
+    let sym = object_layout.object.symbol(symbol_index)?;
+
+    let resolution = if let Some(resolution) =
+        layout.merged_symbol_resolution(object_layout.symbol_id_range.input_to_id(symbol_index))
+    {
+        resolution
+    } else if let Some(section_index) = object_layout.object.symbol_section(sym, symbol_index)? {
+        // TODO: remove
+        let section_name = String::from_utf8_lossy(
+            object_layout
+                .object
+                .section_name(object_layout.object.section(section_index)?)?,
+        );
+        tracing::debug!(%section_name);
+        object_layout.section_resolutions[section_index.0].unwrap_or({
+            // TODO: it's likely a string merge section
```

Comment:
@davidlattimore  Where should I grab the resolution (other than manually constructing that) for the string merge debug info sections?

Reply:
Can you call Resolution::value_with_addend, or alternatively directly call get_merged_string_output_address? For at least some relocations, we're already computing offsets into string-merge sections at ELF writing time, and that happens via those two functions

Reply:
I guess probably `get_merged_string_output_address`, since you don't have a resolution - which makes sense, since non-debug relocations wouldn't have referenced debug strings, so they wouldn't have resolutions.

Reply:
All right, I've just rewritten `apply_debug_relocation` function where I use the suggested function.